.github/workflows/e2e-test-tf-module.yml

```diff
@@ -0,0 +1,193 @@
+name: e2e test Terraform module
+
+on:
+  workflow_dispatch:
+    inputs:
+      ref:
+        type: string
+        description: "Git ref to checkout"
+        required: false
+      cloudProvider:
+        description: "Which cloud provider to use."
+        type: choice
+        options:
+          - "aws"
+          #- "gcp"
+          #- "azure"
+        default: "aws"
+        required: true
+      region:
+        description: "Which region to use."
+        type: choice
+        options:
+          - "eu-west-1"
+          #- "us-east-2" # might want to add later, but would need to specify node groups settings explicitly then
+        default: "eu-west-1"
+        required: true
+      zone:
+        description: "Which zone to use."
+        type: choice
+        options:
+          - "eu-west-1b"
+          #- "us-east-2c"
+        default: "eu-west-1b"
+        required: true
+      image:
+        description: "Node image version of the cluster."
+        type: string
+        default: "v2.12.0"
+        required: true
+      name:
+        description: "Name of the cluster."
+        type: string
+        default: "e2eTfMod"
+        required: true
+  workflow_call:
+    inputs:
+      ref:
+        type: string
+        description: "Git ref to checkout"
+        required: false
+      cloudProvider:
+        description: "Which cloud provider to use."
+        type: string
+        default: "aws"
+      region:
+        description: "Which region to use."
+        type: string
+        default: "eu-west-1" # TODO switch to us-east-2?
+      zone:
+        description: "Which zone to use."
+        type: string
+        default: "eu-west-1b" # TODO "us-east-2c"?
+      image:
+        description: "Node image version of the cluster."
+        type: string
+        default: "v2.12.0"
+        required: true
```

Comment:
We should use an image reference here, similar to the other e2e tests. With that, we can ensure image consistency between the runs in an, e.g., weekly test.